index.ts

```diff
@@ -822,6 +876,51 @@ class IntegrationAPI extends EventEmitter {
   public updateEntityAttributes(entityId: string, attributes: { [key: string]: string | number | boolean }): boolean {
     return this.#configuredEntities.updateEntityAttributes(entityId, attributes);
   }
+
+  public async generateOauth2AuthUrl(): Promise<{ auth_url: string }> {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("generateOauth2AuthUrl: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GenerateOauth2AuthUrl);
+  }
+
+  public async createOauth2Config(data: { [key: string]: any }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("createOauth2Config: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.CreateOauth2Cfg, data);
+  }
+
+  public async deleteOauth2Token(data: { [key: string]: any }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("deleteOauth2Token: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.DeleteOauth2Token, data);
+  }
+
+  public async getOauth2Token(data: { token_id: string; force_refresh: boolean }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("getOauth2Token: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GetOauth2Token, data);
```

Comment:
The request id in a request message must be increased for every new message. Using a random ID is not correct!

Specification:
https://github.com/unfoldedcircle/core-api/blob/ee10ae69784fad38526fe466ff0e874db0369e73/integration-api/UCR-integration-asyncapi.yaml#L797-L802

This is currently not verified in the core service, but this could change anytime!